### DISCUSSION TITLE
How to use find_element, get_attribute and get_text over iteration of elements after finding the elements?

### DISCUSSION THREAD

**Discussion by breadacker** [seleniumbase/SeleniumBase#3206] [Q&A] [answered]
//<tr class="hidden-item ng-star-inserted"
//        >th class="align-middle">10:15<th
//</tr

time_slot_rows = sb.driver.find_elements('tr.hidden-item.ng-star-inserted')

if time_slot_rows:
        # Build a list of tuples with time and row element
        time_slots = []
        for row in time_slot_rows:
            # Extract the time text from the <th> element
            time_text = row.get_text("th.align-middle")
            //time_text = row.find_element("th.align-middle").text
            //time_text = row.get_attribute("th.align-middle.text")
            
 Does find_elements also include the child classes? If so, why do I keep getting mistakes in these get_text, get_attribute and find_element? I would be glad to be corrected

**Comment by mdmintz** [comment, marked as answer]
When you find elements via `find_elements`, those are Selenium WebElement objects (not SeleniumBase objects). You have to use the regular Selenium API if you have a Selenium WebElement object, which doesn't autodetect the selector type from the selector. In other words, you need to include the `by` arg with your calls, eg: `by="css selector`.

**Reply by breadacker** [reply]
row.find_element(By.CSS_SELECTOR, "th.align-middle") as stated in Selenium library seems to work. Thanks for the hint. Appeciate it <3
